crypto/rand/rand_unix.c

```diff
@@ -123,13 +119,23 @@ size_t RAND_POOL_acquire_entropy(RAND_POOL *pool)
 #  endif
 
 #  if defined(OPENSSL_RAND_SEED_OS)
-#   if defined(DEVRANDOM)
-#    define OPENSSL_RAND_SEED_DEVRANDOM
-#   else
+#   if !defined(DEVRANDOM)
 #    error "OS seeding requires DEVRANDOM to be configured"
 #   endif
+#   define OPENSSL_RAND_SEED_DEVRANDOM
+#   if defined(__GLIBC__)
+#    if defined(__GLIBC_PREREQ)
```

Comment:
The triply-nested if could be collapsed into one.  Side note.

Reply:
Not if you look at the comment in rand_lib.c where I based this on.

Reply:
The `defined(__GLIBC__)` and `defined(__GLIBC_PREREQ)` could be combined into a single #if.  The `__GLIBC_PREREQ(2, 25)` cannot and has to stay separate.  rand_lib.c needed to do something special if `*not* defined(__GLIBC__)` which isn't the case here.